provider/custom_event_source_resource.go

```diff
@@ -0,0 +1,139 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/firehydrant/firehydrant-go-sdk/models/components"
+	"github.com/firehydrant/firehydrant-go-sdk/models/sdkerrors"
+	"github.com/firehydrant/terraform-provider-firehydrant/firehydrant"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceCustomEventSource() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: createResourceCustomEventSource,
+		ReadContext:   readResourceCustomEventSource,
+		UpdateContext: updateResourceCustomEventSource,
+		DeleteContext: deleteResourceCustomEventSource,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"slug": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"javascript": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func createResourceCustomEventSource(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*firehydrant.APIClient)
+
+	description := d.Get("description").(string)
+
+	request := components.CreateSignalsEventSource{
+		Name:        d.Get("name").(string),
+		Slug:        d.Get("slug").(string),
+		Description: &description,
+		Javascript:  d.Get("javascript").(string),
```

Comment:
I need to write somewhere that I'm mad about this, so I'm choosing here.  

So `name` and `slug` are strings, sure that makes sense.  Oh, `description` is a pointer to a string, weird, but I guess I could see it if we're being careful about memory and we think that's going to be too long so we want an efficient way to pass it around or something, fine.  But then `javascript` is just a string again?  When that's arguably longer than the description in most cases?  I guess the code generator doesn't know that, but it feels like stupidly adding extra complexity for no real justification.

Whatever.  Enjoy your pointer, `description`.

Reply:
random pointer immediately followed with an ironic lack of a pointer 🙃 

I wonder how it decided on / arrived at this outcome